src/mysql/connection.cr

```diff
@@ -6,7 +6,7 @@ class MySql::Connection < DB::Connection
     @socket = uninitialized TCPSocket
 
     begin
-      host = context.uri.host.not_nil!
+      raise "no host provided" unless host = context.uri.hostname
```

Comment:
I like the custom error message, but wouldn't be more clean to keep assignments to the left?

```suggestion
      host = context.uri.hostname || raise "no host provided"
```
I use `raise ".." unless condition` or `next unless condition` only when condition has no effect on the state I think.